src/imgui/DebuggableEditor.cc

```diff
@@ -485,6 +488,12 @@ void DebuggableEditor::drawContents(const Sizes& s, Debuggable& debuggable, unsi
 					dataEditingTakeFocus = true;
 				}
 			}
+			auto& shortcut = manager.getShortcuts().getShortcut(ShortcutIndex::GOTO_MEMORY_ADDRESS);
+			auto flags = (shortcut.local ? ImGuiInputFlags_RouteUnlessBgFocused : ImGuiInputFlags_RouteGlobalLow | ImGuiInputFlags_RouteUnlessBgFocused) | (shortcut.repeat ? ImGuiInputFlags_Repeat : 0);
+			// invalid shortcut causes SIGTRAP so we test if keychord is valid
+			if (shortcut.keychord && ImGui::Shortcut(shortcut.keychord, 0, flags)) {
+				ImGui::SetKeyboardFocusHere(-1);
+			}
```

Comment:
This must be simplified. Adding a new shortcut in the code should be a single line (next to extending the enum and table). Maybe something like this:
```
if (manager.getShortcuts().checkShortcut(ShortcutIndex::GOTO_MEMORY_ADDRESS)) {
   ...
}
```
In the implementation of `checkShortcut()` we can then do all the things you did here (lookup keychord/type/repeat, check that the shortcut is active, translate to ImGui flags, call ImGui::Shortcut()).

--

I think it would be useful if you could add a few more examples of shortcuts in your patch. That way we more clearly see how much effort is involved in adding new shortcuts. It would be useful if those examples also demonstrate the local vs global and the repeat feature. And it would also be nice if those features can be edited via the GUI.